src/main/java/com/google/devtools/build/lib/analysis/RepoMappingManifestAction.java

```diff
@@ -196,41 +199,41 @@ public DeterministicWriter newDeterministicWriter() {
                       // All packages in a given repository have the same repository mapping, so the
                       // particular way of resolving duplicates does not matter.
                       (first, second) -> first));
+      if (sortedRepoMappings.isEmpty()) {
+        return;
+      }
       if (emitCompactRepoMapping) {
-        Entry<RepositoryName, RepositoryMapping> lastEntry = null;
-        boolean reused = false;
-        ImmutableSet<Entry<RepositoryName, RepositoryMapping>> sentinel =
-            ImmutableSet.of(
-                Maps.immutableEntry(
-                    RepositoryName.MAIN,
-                    // The entries of this repo mapping never equal any real repo mapping's entries
-                    // (never empty).
-                    RepositoryMapping.create(ImmutableMap.of(), RepositoryName.MAIN)));
-        for (var repoAndMapping : Iterables.concat(sortedRepoMappings.entrySet(), sentinel)) {
-          // Module extension repos (and only those repos) all have reference equal repo mapping
-          // entries due to ModuleExtensionRepoMappingEntriesFunction. We only emit these identical
-          // mappings once, with a prefix match for the source repo name.
-          if (lastEntry != null
-              && repoAndMapping.getValue().entries() == lastEntry.getValue().entries()) {
-            reused = true;
+        var repoAndMappings = sortedRepoMappings.entrySet().iterator();
+        var repoAndMapping = repoAndMappings.next();
```

Comment:
looks to me like there's a bug here -- if `repoAndMappings` has only one element, then nothing is output.